### PR TITLE
stats/opencensus: Handle `PickerUpdated` to avoid "Received unexpected stats" error

### PR DESCRIPTION
**Pull request by townba**
RELEASE NOTES: None

### CODE REVIEW

File: stats/opencensus/stats.go [hunk 1]
```diff
@@ -122,10 +122,11 @@ func recordRPCData(ctx context.Context, s stats.RPCStats, mi *metricsInfo) {
 		return
 	}
 	switch st := s.(type) {
-	case *stats.InHeader, *stats.OutHeader, *stats.InTrailer, *stats.OutTrailer:
+	case *stats.InHeader, *stats.OutHeader, *stats.InTrailer, *stats.OutTrailer, *stats.PickerUpdated:
 		// Headers and Trailers are not relevant to the measures, as the
 		// measures concern number of messages and bytes for messages. This
 		// aligns with flow control.
+		// Updated pickers are not relevant to the measures, either.
```
Comment (zasweq) [inline stats/opencensus/stats.go:129]: Nit: please wrap to 80 chars, as above.

**Review by zasweq — APPROVED** [review]
Thanks for the PR, LGTM.

File: stats/opencensus/stats.go (hunk 1 above)
Comment (dfawley) [reply stats/opencensus/stats.go:129]: I would prefer to make the whole comment flow better: "Headers, Trailers, and picker updates are not relevant......"

Comment (townba) [reply stats/opencensus/stats.go]: Done.

**Comment by townba** [comment]
PTAL

**Pull request merged**